packages/template/src/Adaptor.js

```diff
@@ -2,9 +2,57 @@ import {
   execute as commonExecute,
   composeNextState,
 } from '@openfn/language-common';
-import { expandReferences } from '@openfn/language-common/util'
-import { request } from './Utils';
 
+import {
+  request as commonRequest,
+  expandReferences,
+} from '@openfn/language-common/util';
+
+import { makeBasicAuth, assertConfiguration } from './Utils';
+
+export function request(method, path, params, callback) {
```

Comment:
I'm not sure I like this pattern. I guess we need to be thinking about what we want to do with the template.

I don't like this code because it's full of boiler-plate. Repetitive code that doesn't add value.

I understand why we expand references - that's important and good.

I like the assertConfiguration pattern (although I wonder if `validateConfig` or `checkConfig` would be better names (yeah, I know I've asked you to use `assert` before, but I'm not sure it's so helpful in this context)).

I'm not keen on any of header handling - all it should really need to do is auth. Doesn't the common util handle JSON for us?

I'm also not keen on the abstraction. This adaptor has  `create` and a `request` API, where `create` is just a wrapper which does nothing. It isn't at all clear why both functions need to exist - if this is a complete adaptor, you'd only need one.

I would rather have `create` and `get` functions, designed to get and create resources. Maybe they use a common `request` but I think it would establish a more helpful template pattern if each operation handled its own references, called a helper function to add auth, and then called common request (which then means you don't need to alias common request).

As a philosophical point, I don't mind at all if each operation function handles its own references, even if they're copied and pasted. I might even prefer that pattern because it's clear and explicit: each operation handles its own inputs, then passes them off to internal helpers.